osquery/tables/system/posix/yum_sources.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+namespace osquery {
+namespace tables {
+
+const std::string kYumConf { "/etc/yum.conf" };
+const std::string kYumReposDir { "/etc/yum.repos.d" };
+
+void parseYumConf(const std::string& source, QueryData& results, std::string& repos_dir) {
+  // Default value
+  repos_dir = kYumReposDir;
+
+  std::string content;
+  if (!readFile(source, content)) {
+    return;
+  }
+
+  std::string section = "";
+  for (const auto& line : osquery::split(content, "\n")) {
+    auto size = line.size();
+
+    // Skip trivial lines
+    if (size <= 2) {
+      continue;
+    }
+
+    // Skip comments.
+    if (line[0] == '#') {
+      continue;
+    }
+
+    // Section name
+    if  (line[0] == '[' && line[size - 1] == ']') {
+      section = line.substr(1, size - 2);
+      continue;
+    }
+
+    // Skip option-like lines outside of a section
+    if ("" == section) {
+      continue;
+    }
+
+    // Options
+    auto pos = line.find("=");
+    if (pos == std::string::npos) {
+      continue;
+    }
+
+    std::string option = line.substr(0, pos);
+    std::string value = line.substr(pos + 1, size - pos - 1);
+    if ("main" == section) {
```

Comment:
yum also supports **include** syntax to include configuration files outside of /etc/yum...

Reply:
Should I deal with the niche case where the include files are recursive somehow?

Reply:
:+1:

Reply:
@spasam now that I use ini_parser this might not be possible anymore. The include injects the file where it's referred to.

It also accepts arbitrary URL so it might not be a good idea to fetch.

Reply:
Yeah, it was just a suggestion. I don't think include is a common practice. Nice work!